src/middleware/authentication.ts

```diff
@@ -0,0 +1,22 @@
+import { AuthenticationError } from "apollo-server";
+import { logger } from "../logger";
+import * as dotenv from "dotenv";
+
+dotenv.config();
+
+export const authenticationContextMiddleware = ({ req }: { req: any }) => {
+  const authHeaderUser: string = req.headers.authorization || "";
+  const expectedHeader: string = process.env.AUTH_TOKEN!;
+
+  if (authHeaderUser !== expectedHeader) {
+    logger.error(`Unauthorized access attempt: ${req.method} ${req.url}`);
```

Comment:
I wouldn't really log this. Throwing the exception is good enough.

Reply:
Removed it, thanks!